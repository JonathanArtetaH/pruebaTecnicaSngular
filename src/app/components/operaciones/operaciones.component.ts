import { Component, EventEmitter, Input, Output, SimpleChanges } from '@angular/core';

@Component({
  selector: 'app-operaciones',
  template: '', 
})
export class OperacionesComponent {
  @Output() resultadoNumero = new EventEmitter<number>();
  @Input() numeroInput: number  = 0;  
  resultado : number =0; 


  ngOnChanges(changes: SimpleChanges) {  
    if (changes['numeroInput']){ 
      Promise.resolve().then(()=> this.realizarOperacionSerie())
    } 
  }

realizarOperacionSerie(){ 
  let resultadoNumPrimo = this.calcularNumeroPrimo(this.numeroInput) 
  let resultadoTriangular = this.obtenerTriangular(this.numeroInput)
  let resultadoNumFibonacci = this.calcularNumeroFibonacci(this.numeroInput) 

  this.resultado  = this.caculcarSeriesNum(resultadoNumPrimo,resultadoTriangular,resultadoNumFibonacci) 
  this.emitirResultNumero()
  
}


caculcarSeriesNum(numPrimo: number, numTriangular: number, numFibonacci: number){ 
  // console.log('Num primo: ',numPrimo,'\nNum triangular:',numTriangular,'\nNum fibo:' ,numFibonacci) 
  return  ( 2 * numPrimo ) * ( 3 * numTriangular ) * ( 7 * numFibonacci );  
}

// empieza calcular fibonacci
calcularNumeroFibonacci(n: number){
  let resultRestNum = n -1;
  if(resultRestNum == 0) return 0; 

  let resultFun = 0;
  let numFunUno = 0;
  let numFunDos = 1;

  for (let i = 2; i <= resultRestNum; i++) {
    resultFun = numFunUno + numFunDos;
    numFunUno = numFunDos;
    numFunDos = resultFun;    
  }

  return resultFun;
}
// termina calcular fobinacci

// Empieza calcular numero primo
calcularNumeroPrimo(n: number) {  
  let numeroTotal = n - 2;
  let i = 2;
  let arraNumsPrimos = [];
  if(numeroTotal<2) return 0; 
   while (arraNumsPrimos.length < numeroTotal) { 
      if (this.esPrimo(i)) arraNumsPrimos.push(i);
      i++;
  }
  return arraNumsPrimos[numeroTotal - 1];
}

esPrimo(num: number) {  
  for (let i = 2; i <= Math.sqrt(num); i++) {
      if (num % i === 0) return false; 
  }
  return num > 1;
}
// termina calcular numero primo


// empieza calcular triangular
obtenerTriangular(n: number) { 
  let numRestado = (n - 2 );
  if(numRestado >=1  ) return ( numRestado * ( numRestado +1 ) ) / 2 
  return 0;
} 
// termina calcular triangular
 
emitirResultNumero(){
    this.resultadoNumero.emit(this.resultado);
}
 
}
