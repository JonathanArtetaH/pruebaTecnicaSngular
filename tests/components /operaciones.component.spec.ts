import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';    
import { OperacionesComponent } from '../../src/app/components/operaciones/operaciones.component';

describe('OperacionesComponent', () => {
  
  let component: OperacionesComponent;
  let fixture: ComponentFixture<OperacionesComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        OperacionesComponent
      ],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(OperacionesComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('Creacion de componente correctamente', () => {
    expect(component).toBeTruthy();
  });

  test('Debe coincidir con el snapshot', () => {
    const fixture = TestBed.createComponent(OperacionesComponent);
    fixture.detectChanges();
    const compiled = fixture.nativeElement as HTMLElement;
    expect( compiled ).toMatchSnapshot();
  })
   
});
