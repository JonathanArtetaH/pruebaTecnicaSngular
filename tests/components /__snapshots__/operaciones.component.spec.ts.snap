// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperacionesComponent Debe coincidir con el snapshot 1`] = `
<div
  id="root2"
/>
`;
